### PR TITLE
fix(artifact test): verify that 'housekeeping' attribute exists

### PR DESCRIPTION
**Pull request by juliayakovlev**
Task:
https://trello.com/c/v204rFpm/4339-attributeerror-artifactstest-object-has-no-attribute-housekeeping

Failed test:
https://jenkins.scylladb.com/job/scylla-master/job/artifacts/job/artifacts-ubuntu2004-arm-test/61/execution/node/52/log/

Test failed because failed to create the node and actually the tes was not started.
As result the 'housekeeping' attribute wasn't created.
The fix is to check if the attribute exists

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Review by dkropachev — CHANGES_REQUESTED** [review]
Could you please also fix HousekeepingDB.close to do not fail if no HousekeepingDB.connect was called

**Comment by juliayakovlev** [comment]
> Could you please also fix HousekeepingDB.close to do not fail if no HousekeepingDB.connect was called

Done

File: artifacts_test.py [hunk 1]
```diff
@@ -37,7 +37,8 @@ def setUp(self) -> None:
         self.housekeeping.connect()
 
     def tearDown(self) -> None:
-        self.housekeeping.close()
+        if getattr(self, "housekeeping"):
```
Comment (enaydanov) [inline artifacts_test.py:40]:
```
Help on built-in function getattr in module __builtin__:

getattr(...)
    getattr(object, name[, default]) -> value
    
    Get a named attribute from an object; getattr(x, 'y') is equivalent to x.y.
    When a default argument is given, it is returned when the attribute doesn't
    exist; without it, an exception is raised in that case.
```

It should be `getattr(self, "housekeeping", None)`

**Review by enaydanov — CHANGES_REQUESTED** [review]

File: artifacts_test.py (hunk 1 above)
Comment (juliayakovlev) [reply artifacts_test.py]: fixed

**Review by dkropachev — APPROVED** [review]

**Review by enaydanov — APPROVED** [review]

**Pull request merged**